### PR TITLE
{phys}[foss/2023a] BAGEL v1.2.2

### PR DESCRIPTION
**Pull request by TopRichard**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch [hunk 1]
```diff
@@ -0,0 +1,88 @@
+diff -ruN bagel-1.2.2/src/molecule/shell_base.cc bagel-1.2.2.orig/src/molecule/shell_base.cc
```
Comment (verdurin) [inline easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch:1]: @TopRichard could you add a comment here explaining the purpose of the patch please?

**Review by verdurin — DISMISSED** [review]

File: easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch (hunk 1 above)
Comment (TopRichard) [reply easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch:1]: The patch is to resolve missing boost includes and fix ambiguous function call, similar to https://github.com/qsimulate-open/bagel/pull/229

Comment (smoors) [reply easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch:2]: @TopRichard please move the comment to the top of the patch (above the diff line) and also add the patch author.
also, don't forget to update the checksum of the patch.

Comment (TopRichard) [reply easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2.patch:4]: Done

File: easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+source_urls = ['https://github.com/nubakery/bagel/archive/']
+sources = ['v%(version)s.tar.gz']
+
+patches = ['%(name)s-%(version)s.patch']
+
+checksums = [
+    {'v1.2.2.tar.gz': 'b9ef9ad5ad9b836b5393badb5a3e5b162d6ff32da497f69d05ef9e4d091b558a'},
+    {'BAGEL-1.2.2.patch': '6a0bb2b0c762626219b9216cd96d287e7c14af72234e7ec40c5d684010747792'},
```
Comment (smoors) [inline easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2-foss-2023a.eb:22]: checksum is still not correct.
simplest way to fix this:
```
eb BAGEL-1.2.2-foss-2023a.eb --inject-checksums --force
```

Comment (TopRichard) [reply easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2-foss-2023a.eb]: When updating the PR after injecting the new checksum, it is deleting the main BAGEL.eb file.

Comment (TopRichard) [reply easybuild/easyconfigs/b/BAGEL/BAGEL-1.2.2-foss-2023a.eb]: Had issues with pushing the updated files, but looks fine now.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@boegelbot please test @ jsc-zen2

**Comment by smoors** [comment]
Going in, thanks @TopRichard!

**Pull request merged**